### DISCUSSION TITLE
do we have plan about id link?

### DISCUSSION THREAD

**Discussion by WaytrueDeng** [nvim-orgmode/orgmode#500] [General]
id link is a great feature which heavily used in org-roam, and it is easy to navigate to other file heading by id even after we change the file path.

may I ask do we have a plan to implement this feature?

thanks a lot for devs' hardworking

**Comment by jgollenz** [comment]
Hi @WaytrueDeng, is that a part of vanilla orgmode or is it org-roam functionality. If it's the former: please drop a link and I'll have a look at it :+1: If it's the latter, then the answer is probably no. For now, we are only focusing on porting orgmode

**Reply by sg1fan** [reply]
I believe this functionality is implemented in [org-id.el](https://git.savannah.gnu.org/cgit/emacs/org-mode.git/tree/lisp/org-id.el) for org-mode.  In that file, we can see the various configuration variables. For example the variable `org-id-locations-file` can be set to a specific file, which will contain the id:filepath mappings. 

org-roam simply forces all linkable targets (which they call nodes) to be associated with an id, which is then stored in sqlite (instead of a plaintext file) along with its aliases so it can query for backlinks and aliases. However, if you suddenly turned off org-roam, you would still be able to traverse these id links with vanilla org-mode.

**Reply by jgollenz** [reply]
@sg1fan thank you for the explanation. I haven't looked at it in depth, but in general it looks like a feature that should be in nvim-orgmode as well. I'll convert this discussion to an issue

**Comment by aareman** [comment]
This plugin exists https://github.com/chipsenkbeil/org-roam.nvim

**Reply by kristijanhusak** [reply]
There is also built in functionality for ID links https://github.com/nvim-orgmode/orgmode/blob/master/DOCS.md#org_store_link
